test/gnosisSafePersonalEdition.js

```diff
@@ -49,10 +53,61 @@ contract('GnosisSafePersonalEdition', function(accounts) {
+        await safeUtils.executeTransaction(lw, gnosisSafe, 'executeTransaction withdraw 0.5 ETH', [lw.accounts[0], lw.accounts[2]], accounts[0], web3.toWei(0.5, 'ether'), "0x", CALL, executor, {
+          gasToken: token.address
+        });
+
+        await safeUtils.executeTransaction(lw, gnosisSafe, 'executeTransaction withdraw 0.5 ETH', [lw.accounts[0], lw.accounts[2]], accounts[0], web3.toWei(0.5, 'ether'), "0x", CALL, executor, {
+          gasToken: token.address
+        })
+        let executorDiff = (await token.balances(executor)).toNumber() - executorBalance;
+        console.log("    Executor earned " + executorDiff + " Tokens")
+        assert.ok(executorDiff > 0);
+    });
+
+    it('should fail when depositing 1 ETH paying with token due to token transfer fail', async () => {
+        let mockContract = await MockContract.new();
+        let mockToken = MockToken.at(mockContract.address);
+        let transferData = await mockToken.contract.transfer.getData(executor, 91861);
+        await mockContract.givenRevertAny(transferData.slice(0,10));
+        await web3.eth.sendTransaction({from: accounts[0], to: gnosisSafe.address, value: web3.toWei(1.1, 'ether')})
+        await utils.assertRejects(
+            safeUtils.executeTransaction(lw, gnosisSafe, 'executeTransaction withdraw 0.5 ETH', [lw.accounts[0], lw.accounts[2]], accounts[0], web3.toWei(0.5, 'ether'), "0x", CALL, executor, { gasToken: mockToken.address }),
+            "ERC20 token transfer reverted"
+        );
+
+        mockContract = await MockContract.new();
+        mockToken = MockToken.at(mockContract.address);
+        transferData = await mockToken.contract.transfer.getData(executor, 91861);
+        await mockContract.givenOutOfGasAny(transferData.slice(0,10));
+        await web3.eth.sendTransaction({from: accounts[0], to: gnosisSafe.address, value: web3.toWei(1.1, 'ether')})
+        await utils.assertRejects(
+            safeUtils.executeTransaction(lw, gnosisSafe, 'executeTransaction withdraw 0.5 ETH', [lw.accounts[0], lw.accounts[2]], accounts[0], web3.toWei(0.5, 'ether'), "0x", CALL, executor, { gasToken: mockToken.address }),
+            "ERC20 token transfer out of gas"
+        );
+
+        mockContract = await MockContract.new();
+        mockToken = MockToken.at(mockContract.address);
+        transferData = await mockToken.contract.transfer.getData(executor, 91861);
+        await mockContract.givenReturnAny(transferData.slice(0,10), abi.rawEncode(['bool'], [false]).toString());
+        await web3.eth.sendTransaction({from: accounts[0], to: gnosisSafe.address, value: web3.toWei(1.1, 'ether')})
+        await utils.assertRejects(
+            safeUtils.executeTransaction(lw, gnosisSafe, 'executeTransaction withdraw 0.5 ETH', [lw.accounts[0], lw.accounts[2]], accounts[0], web3.toWei(0.5, 'ether'), "0x", CALL, executor, { gasToken: mockToken.address }),
+            "ERC20 token transfer returned false"
```

Comment:
This is the message shown when the test fails, so probably something like `Transaction should fail if the ERC20 token transfer method returns false` is more helpful ;)

Reply:
Same for the others 😉

Reply:
I have made the messages more meaningful